MainProject/Models/Transformator.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            AngleRadZ += angleRad;
+            SinZ = (float)Math.Sin(AngleRadZ);
+            CosZ = (float)Math.Cos(AngleRadZ);
+            if (update) UpdateTransformation();
+        }
+
+        public void Rotate(Vector angles)
+        {
+            RotateX(angles.X, false);
+            RotateY(angles.Y, false);
+            RotateZ(angles.Z, true);
+        }
+
+        public void MoveX(float directionX)
+        {
+            ShiftX += directionX;
+            UpdateTransformation();
+        }
+
+        public void MoveY(float directionY)
+        {
+            ShiftY += directionY;
+            UpdateTransformation();
+        }
+
+        public void MoveZ(float directionZ)
+        {
+            ShiftZ += directionZ;
+            UpdateTransformation();
+        }
+
+        public void Move(Vector direction)
+        {
+            ShiftX += direction.X;
+            ShiftY += direction.Y;
+            ShiftZ += direction.Z;
+            UpdateTransformation();
+        }
+
+        public void ToScaleX(float scaleX)
+        {
+            ScaleX *= scaleX;
+            UpdateTransformation();
+        }
+
+        public void ToScaleY(float scaleY)
+        {
+            ScaleY *= scaleY;
+            UpdateTransformation();
+        }
+
+        public void ToScaleZ(float scaleZ)
+        {
+            ScaleZ *= scaleZ;
+            UpdateTransformation();
+        }
+
+        public void ToScale(Vector scale)
+        {
+            ScaleX *= scale.X;
+            ScaleY *= scale.Y;
+            ScaleZ *= scale.Z;
+            UpdateTransformation();
+        }
+
+        private void UpdateTransformation()
+        {
+            var Rotate = new float[4, 4]
+            {{ CosY * CosZ,                      -SinZ * CosY,                      SinY,         0 },
+             { SinX * SinY * CosZ + SinZ * CosX, -SinX * SinY * SinZ + CosX * CosZ, -SinX * CosY, 0 },
+             { SinX * SinZ - SinY * CosX * CosZ,  SinX * CosZ + SinY * SinZ * CosX,  CosX * CosY, 0 },
+             { 0,                                                               0,            0,  1 }};
+
+            var Scale = new float[4, 4]
+            {{ ScaleX, 0, 0, 0 },
+             { 0, ScaleY, 0, 0 },
+             { 0, 0, ScaleZ, 0 },
+             { 0, 0, 0, 1 }};
+
+            var Shift = new float[4, 4]
+            {{ 1, 0, 0, ShiftX },
+             { 0, 1, 0, ShiftY },
+             { 0, 0, 1, ShiftZ },
+             { 0, 0, 0, 1 }};
+
+            TransformationMatrix = Multiply(Multiply(Shift, Rotate), Scale);
+        }
+
+        //public void ResetTransformation()
+        //{
+        //    ShiftX = 0;
+        //    ShiftY = 0;
+        //    ShiftZ = 0;
+        //    ScaleX = 1;
+        //    ScaleY = 1;
+        //    ScaleY = 1;
+        //    RotateX(-AngleRadX, false);
+        //    RotateY(-AngleRadY, false);
+        //    RotateZ(-AngleRadZ, true);
+        //}
+
```

Comment:
let's remove unused code

Reply:
I forgot to delete before pushing, it is already done